### PR TITLE
Feature student on-boarding tour

### PR DESCRIPTION
**Pull request by pandafulmanda**
https://trello.com/c/cbmPWBUP/1064-change-request-have-onboarding-tour-for-students

We want to create a short onboarding tour just for students that covers: 
- [x] This is your dashboard
- [x] This is your performance forecast
- [x] This is where you can browse the book
- [x] This is your Getting Started Guide

We want to also add contextual training wheels for:
- [x] late assignments
- [x] highlighting 
- [x] HW feedback 

screen shot 2017-12-14 at 12 29 52 pm
screen shot 2017-12-14 at 2 03 48 pm
screen shot 2017-12-14 at 2 06 48 pm
screen shot 2017-12-14 at 2 06 52 pm
screen shot 2017-12-14 at 2 06 56 pm

## Contextual

screen shot 2017-12-21 at 1 18 13 pm
screen shot 2017-12-21 at 2 26 03 pm
screen shot 2017-12-14 at 2 02 29 pm
screen shot 2017-12-14 at 2 01 11 pm

### CODE REVIEW

File: tutor/src/components/annotations/summary-toggle.jsx [hunk 1]
```diff
@@ -38,12 +39,14 @@ export default class AnnotationSummaryToggle extends React.Component {
...
-      </div>
+      <TourAnchor id="student-highlighting-button">
+        <div className="annotation-summary-toggle">
+          <Icon
+            type="pencil-square-o"
+            onClick={User.annotations.ux.toggleSummary}
+          />
+        </div>
+      </TourAnchor>
     );
```
Comment (nathanstitt) [inline tutor/src/components/annotations/summary-toggle.jsx]: You could let the `TourAnchor` act as the wrapper div by setting the `className` on it, such as:
```
  <TourAnchor id="student-highlighting-button" className="annotation-summary-toggle">
    <Icon type="pencil-square-o"  onClick={User.annotations.ux.toggleSummary} />
  </TourAnchor>
```

File: tutor/src/models/annotations.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ export default class Annotations extends Map {
   @action.bound updateAnnotations(annotations) {
     this.api.requestsInProgress.delete('fetch');
     this.api.requestCounts.read += 1;
-    if (annotations) {
+    if (isArray(annotations)) {
       annotations.forEach((an) => {
```
Comment (nathanstitt) [inline tutor/src/models/annotations.js]: Cool, glad you fixed that 😁

File: tutor/specs/components/navbar/__snapshots__/support-menu.spec.jsx.snap [hunk 3]
```diff
@@ -150,22 +155,28 @@ exports[`Support Menu renders support links when in a course for student 1`] = `
...
-        onClick={[Function]}
-        onKeyDown={[Function]}
-        role="menuitem"
-        tabIndex="-1"
-        target="_blank"
+      <li
+        className="support-document-link"
+        role="presentation"
+        style={undefined}
       >
```
Comment (nathanstitt) [inline tutor/specs/components/navbar/__snapshots__/support-menu.spec.jsx.snap]: It looks to me like you've got an incorrect DOM here,  it's nesting as:
```
<ul>
    <div>
          <li>
```


I found where this is happening and noted it below

File: tutor/src/components/navbar/support-menu.jsx [hunk 4]
```diff
@@ -122,7 +122,9 @@ class SupportMenuDropDown extends React.PureComponent {
         >
           <span>Help Articles</span>
         </MenuItem>
-        <SupportDocument courseId={courseId} />
+        <TourAnchor id="menu-support-document">
+          <SupportDocument courseId={courseId} />
+        </TourAnchor>
         <MenuItem
```
Comment (nathanstitt) [inline tutor/src/components/navbar/support-menu.jsx]: The snapshots suggest that we've got an incorrect DOM here,  it's nesting as:
```html
<ul>
    <div>
          <li>
```

I think if we add the  `tagName="li" prop to `TourAnchor` it should fix this, but then `SupportDocument` will need updated to not have a `li` tag.

```jsx

<ul>
   <TourAnchor tagname="li" id="menu-support-document" className="support-document-link">
       <a href="https://d3bxy9euw4e147.cloudfr/openstax_tutor_student_getting_s" ... />
   </TourAnchor>
</ul>

```

Comment (pandafulmanda) [reply tutor/src/components/navbar/support-menu.jsx]: Oh thanks for catching and fixing!

**Pull request merged**